game-core/src/main/java/games/strategy/engine/framework/startup/launcher/ServerLauncher.java

```diff
@@ -187,44 +175,25 @@ void launchInternal(final Component parent, @Nullable final Void none) {
       });
       stopGame();
     } catch (final RuntimeException e) {
-      final String errorMessage = "Unrecognized error occurred. If this is a repeatable error, "
-          + "please make a copy of this savegame and report to:\n" + UrlConstants.GITHUB_ISSUES;
-      log.log(Level.SEVERE, errorMessage, e);
-      stopGame();
+      // no-op, this is a simple player disconnect, no need to scare the user with some giant stack trace
+      if (!(e.getCause() != null && e.getCause() instanceof ConnectionLostException)) {
```

Comment:
Is the null check needed?

Reply:
Probably not, but I wanted to be safe here.

Never underestimate null. ^^

Reply:
Likely indicates a missing test case. We should be more certain than not, the `!(!a && b)` conditional is a bit confusing compared to `a || !b`. Simplest way I see to resolve is to remove the null check.